### PR TITLE
[XPU] Add bf16 tolerance override for LinearCrossEntropyLoss test_save_load

### LINKED ISSUE #193407: DISABLED test_save_load_nn_LinearCrossEntropyLoss_xpu_bfloat16 (__main__.TestModuleXPU)
Platforms: xpu

This test was disabled because it is failing on main branch ([recent examples](https://github.com/pytorch/pytorch/actions/runs/30083005504/job/89453199227#step:13:6124)).

cc @malfet @pytorch/pytorch-dev-infra @gujinghui @EikanWang @fengyuan14

### PR DESCRIPTION
**Pull request by xuhancn**
Fixes #193407

Re-enables the flaky `test_save_load_nn_LinearCrossEntropyLoss_xpu_bfloat16` by adding the XPU bfloat16 tolerance override for `test_save_load` on `nn.LinearCrossEntropyLoss` that CUDA already carries.

## Why

`#193407` disabled this test because `test_save_load` was flaky: it asserts a module and its pickled copy produce equal outputs, but two identical bfloat16 forward passes sporadically differ past the default `rtol=0.016`.

Root cause: for K-dimensional loss (`out_features != ()`), the reference path runs `F.cross_entropy` on bf16 logits, which dispatches to `nll_loss2d_forward_xpu`. That kernel (a faithful port of the upstream CUDA `NLLLoss2d.cu`) reduces across batch blocks by casting the per-block fp32 group-reduction back to bf16 and `atomicAdd`-ing into a single bf16 scalar. The bf16 atomic accumulation order varies run to run, so two identical forward passes yield slightly different results. The kernel calls `alertNotDeterministic()` for `sum`/`mean` reductions — the nondeterminism is upstream-documented and shared with CUDA. The 1-D `nll_loss_forward_xpu` path is deterministic (single in-group reduce, no cross-block atomicAdd).

CUDA already carries a loose tolerance override for this exact test/dtype (`atol=2e-1, rtol=5e-2`); XPU only had the `float16` override, so XPU bf16 fell back to the default `rtol=0.016`, which the ~0.015 accumulation variance just barely exceeds. This adds the matching XPU bfloat16 override — restoring parity with CUDA's established handling rather than diverging XPU into a deterministic-kernel rewrite no backend (including CUDA) has adopted for `nll_loss2d`.

## What changed

`torch/testing/_internal/common_modules.py` — one `DecorateInfo` matching the existing CUDA override, immediately after it, with an explanatory comment.

> The summary, root-cause analysis, and test plan below were drafted with an AI assistant (Claude Code) and reviewed by me. I verified the root cause by reproducing the nondeterminism on a local XPU nightly build and confirming the test passes 10/10 runs with the override vs 6/8 failures without it.

## Test plan

On an XPU nightly build (`2.15.0.dev20260812+xpu`):

```
python test/test_modules.py \
  TestModuleXPU.test_save_load_nn_LinearCrossEntropyLoss_xpu_bfloat16
```

- Without override: 6/8 runs `FAILED` (flaky, reproduces CI failure).
- With override: 10/10 runs `PASS`.

Max observed rel diff over 40 seeds × 20 runs was ~0.0146, well within `atol=2e-1, rtol=5e-2`.

cc @malfet @guangyey @gujinghui @EikanWang @fengyuan14

### CODE REVIEW

**Review by guangyey — APPROVED** [review]

File: torch/testing/_internal/common_modules.py [hunk 1]
```diff
@@ -4735,6 +4735,15 @@ def module_error_inputs_torch_nn_LogSoftmax(module_info, device, dtype, requires
                                 "test_forward", dtypes=[torch.bfloat16]),
                    DecorateInfo(toleranceOverride({torch.bfloat16: tol(atol=2e-1, rtol=5e-2)}), "TestModule",
                                 "test_save_load", device_type="cuda", dtypes=[torch.bfloat16]),
+                   # nll_loss2d_forward_xpu reduces across batch blocks via atomicAdd
```
Comment (guangyey) [inline torch/testing/_internal/common_modules.py]: Nit: The comment is too long

**Review by jansel — APPROVED** [review]
